### PR TITLE
Add /order endpoint similar to SOR but returns a transaction

### PR DESCRIPTION
**Pull request by timjrobinson**
Add /order endpoint + lambda. This takes a SorRequest and returns a SorOrderResponse which is a transaction (to + data + value) the user can immediately post to chain.

The benefit of this is we can return a batchSwap or relayer transaction the user can immediately post to chain without needing to know how to assemble it themselves from the SOR return data. 

This will allow Cowswap to handle join/exit swaps with Balancer without having to re-code their solver to convert a SOR response to different transaction types. 

Todo:
- [x] Full testing with dev account
- [x] Add more unit tests
- [x] Add E2E tests

### CODE REVIEW

File: src/modules/sor/order.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    price: swapInfo.marketSp,
+  };
+
+  return {
+    price: priceResponse,
+    to: ADDRESSES[networkId].contracts.vault,
+    data: encodedBatchSwapData,
+    value: '0',
+  };
+}
```
Comment (garethfuller) [inline src/modules/sor/order.ts:43]: Is the plan to get this endpoint live first, then implement join/exit support?

**Review by garethfuller — APPROVED** [review]

**Comment by timjrobinson** [comment]
Added E2E tests for orders and they are working correctly.

**Comment by timjrobinson** [comment]
Added ability to specify slipapge and a different recipient for your order.

File: src/modules/sor/batch-swap.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    swaps: swapInfo.swaps,
+    assets: swapInfo.tokenAddresses,
+    funds: {
+      fromInternalBalance: false,
+      sender: senderAddress,
+      recipient: recipientAddress,
+      toInternalBalance: false,
+    },
+    limits: limits,
+    deadline: '999999999999999999',
```
Comment (garethfuller) [inline src/modules/sor/batch-swap.ts:88]: Are we sure the deadline should be infinite?

File: src/modules/sor/order.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    buyAmount: swapInfo.returnAmount,
+    allowanceTarget: ADDRESSES[networkId].contracts.vault,
+    price: swapInfo.marketSp,
+  };
+
+  return {
+    price: priceResponse,
+    to: ADDRESSES[networkId].contracts.vault,
+    data: encodedBatchSwapData,
+    value: '0',
```
Comment (garethfuller) [inline src/modules/sor/order.ts:39]: What if the user is trading ETH for DAI? Would the value not have to be specified here?

**Review by garethfuller — APPROVED** [review]

File: src/modules/sor/batch-swap.ts (hunk 2 above)
Comment (timjrobinson) [reply src/modules/sor/batch-swap.ts:88]: Fixed, made it 60 minutes instead.

File: src/modules/sor/order.ts (hunk 3 above)
Comment (timjrobinson) [reply src/modules/sor/order.ts:39]: True, this doesn't support ETH swaps yet. Will fix in a future PR.

File: src/modules/sor/batch-swap.ts (hunk 2 above)
Comment (garethfuller) [reply src/modules/sor/batch-swap.ts]: Why 60mins? Can we ask John or Mike what this should be?

**Pull request merged**